ext/intl/calendar/calendar.stub.php

```diff
@@ -0,0 +1,284 @@
+    public function isEquivalentTo(IntlCalendar $calendar) {}
+
+    /** @return bool */
+    public function isLenient() {}
+
+    /** @return bool */
+    public function isWeekend(?float $date = null) {}
+
+    /**
+     * @param int|bool $amountOrUpOrDown
+     * @return bool
+     */
+    public function roll(int $field, $amountOrUpOrDown) {}
+
+    /** @return bool */
+    public function isSet(int $field) {}
+
+    /** @return bool */
+    public function set(int $year, int $month, int $dayOfMonth = UNKNOWN, $hour = UNKNOWN, int $minute = UNKNOWN, int $second = UNKNOWN) {}
+
+    /** @return bool */
+    public function setFirstDayOfWeek(int $dayOfWeek) {}
+
+    /** @return bool */
+    public function setLenient(bool $isLenient) {}
+
+    /** @return bool */
+    public function setRepeatedWallTimeOption(int $wallTimeOption) {}
+
+    /** @return bool */
+    public function setSkippedWallTimeOption(int $wallTimeOption) {}
+
+    /** @return bool */
+    public function setTime(float $date) {}
+
+    /**
+     * @param IntlTimeZone|DateTimeZone|string|null $timeZone
+     * @return bool
+     */
+    public function setTimeZone($timeZone) {}
+
+    /** @return DateTime|false */
+    public function toDateTime() {}
+}
+
+class IntlGregorianCalendar extends IntlCalendar
+{
+    public function __construct() {}
+
+    /**
+     * @param IntlTimeZone|DateTimeZone|string|null $timeZone
+     * @return IntlGregorianCalendar|null
+     */
+    public static function createInstance($timeZone = null, ?string $locale = null) {}
+
+    /** @return bool */
+    public function setGregorianChange(float $change) {}
+
+    /** @return float|false */
+    public function getGregorianChange() {}
+
+    /** @return bool */
+    public function isLeapYear(int $year) {}
+}
+
+/**
+ * @param IntlTimeZone|DateTimeZone|string|null $timeZone
+ * @return IntlCalendar|null
+ */
+function intlcal_create_instance($timeZone = null, ?string $locale = null) {}
+
+/** @return Iterator|false */
+function intlcal_get_keyword_values_for_locale(string $key, string $locale, bool $commonlyUsed) {}
+
+function intlcal_get_now(): float {}
+
+function intlcal_get_available_locales(): array {}
+
+function intlcal_get(IntlCalendar $calendar, int $field): int|false {}
+
+function intlcal_get_time(IntlCalendar $calendar): float|false {}
+
+function intlcal_set_time(IntlCalendar $calendar, float $date): bool {}
+
+function intlcal_add(IntlCalendar $calendar, int $field, int $amount): bool {}
+
+/** @param IntlTimeZone|DateTimeZone|string|null $timeZone */
+function intlcal_set_time_zone(IntlCalendar $calendar, $timeZone): bool {}
+
+function intlcal_after(IntlCalendar $calendarObject, IntlCalendar $calendar): bool {}
+
+function intlcal_before(IntlCalendar $calendarObject, IntlCalendar $calendar): bool {}
+
+function intlcal_set(IntlCalendar $calendar, int $year, int $month, int $dayOfMonth = UNKNOWN, $hour = UNKNOWN, int $minute = UNKNOWN, int $second = UNKNOWN): bool {}
```

Comment:
Hmm, as it is now, this function accepts also 5 arbitrary non NULL zvals, and returns NULL (without even any notice) in this case, e.g.
````
var_dump(intlcal_set('this', 'one', 'is', 'completely', 'borked'));
````
I think we should fix this first, before adding stubs. Also, the optional parameters are currently nullable.

Reply:
Thanks for the feedback, I'll have a look :)

Reply:
Thanks! It might be possible to just drop most of that manual arg handling (`variant`), but instead to rely on `zend_parse_method_parameters(ZEND_NUM_ARGS(), object, "Oll|l!l!l!l!", ...)`, and to throw on 4 args, and to do the `co->ucal->set()` at the end of the method by checking the boolean is_null flags for the optional parameters (in reversed order).

Reply:
@cmb69 I already have the implementation locally which doesn't allow `null`s, but I am not sure if supporting nullability is a good idea. For example what should happen in this case: `intl_cal($cal, 1,2, null, null, 5, 6)`? An exception should be thrown?

Reply:
Ah, good catch! Actually, none of the parameters are nullable. Basically, it boils down to parsing `"Oll|llll"` and to disallow `ZEND_NUM_ARGS() == 4`.

Reply:
So far only the trailing arguments can have a `null` value, but they are ignored. That said, I'll commit what I have now.

Reply:
Thanks! LGTM (also the error message regarding variant 4).